src/main/eo/org/eolang/math/number.eo

```diff
@@ -192,72 +192,144 @@
   # Make ^.n power x
   # @todo #21:30 min. Add an ability to raise a number to float power.
   #  `exp` and `ln` objects are needed.  sqr(n,x) = exp(n * ln(x))
-  #
-  # @todo #21:30 min. Improve performance to be able to work with big numbers.
-  #  Should use binary exponentiation. It allows to raise any number to the n'th
-  #  power in O(log n) multiplications
   [x] > power
-    as-float > self-as-float!
-
-    and. > both-are-int!
-      (number x).is-int
-      is-int
-
-    if. > x-as-float!
-      (number x).is-nan > x-is-nan!
+    if. > @
+      or.
+        is-nan.
+          number x > x-as-num
+        is-nan
       nan
-      (number x).as-float
-
-    [y] > powf
-      if. > @
-        y.gt 0.0
-        times.
-          self-as-float
-          ^.powf (y.minus 1.0)
-        1.0
-
-    [y] > powi
-      if. > @
-        y.gt 0
-        times.
-          ^.^.n
-          ^.powi (y.minus 1)
-        1
-
-    switch > @
-      *
-        or.
-          x-is-nan
-          is-nan
-        nan
-      *
-        x-as-float.eq 0.0
-        if.
-          both-are-int
-          1
-          1.0
-      *
-        x-as-float.lt 0.0
-        if.
-          self-as-float.eq 0.0
-          positive-infinity
-          div.
-            1.0
-            powf
-              neg.
-                x-as-float
-      *
-        x-as-float.gt 0.0
-        if.
-          self-as-float.eq 0.0
+      []
+        if. > x-as-float!
+          x-as-num.is-infinite > x-is-inf
```

Comment:
@maxonfjvipon Can `x-is-inf` be constant ?

Reply:
@Graur sure